tests/vt_perf/baremetal.pm

```diff
@@ -0,0 +1,99 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: VT perf testsuite on Baremetal testing
+# Maintainer: James Wang <jnwang@suse.com>
+
+package baremetal;
+
+use warnings;
+use strict;
+use base "opensusebasetest";
+use Mitigation;
+use Utils::Backends 'use_ssh_serial_console';
+use bootloader_setup qw(grub_mkconfig change_grub_config add_grub_cmdline_settings remove_grub_cmdline_settings grep_grub_settings set_framebuffer_resolution set_extrabootparams_grub_conf);
+use ipmi_backend_utils;
+use power_action_utils 'power_action';
+use testapi;
+use utils;
+
+my $syspath         = '/sys/devices/system/cpu/vulnerabilities/';
+my $name            = get_var('VM_NAME');
+my $install_url     = get_var('INSTALL_REPO');
+my $logfile_path    = get_var('VM_INST_LOG');
+my $cpu             = get_var('CPU_FEATURE');
+my $vm_pool         = get_var('VM_POOL');
+my $vm_shares       = get_var('VM_SHARES');
+my $source_host     = get_var("SOURCE_HOSTNAME");
+my $dest_host       = get_var("DEST_HOSTNAME");
+my $subname         = "";
+my $cpu_1           = "";
+my $git_repo_url    = get_required_var("MITIGATION_GIT_REPO");
+my $git_user        = get_required_var("MITIGATION_GIT_REPO_USER");
+my $git_pass        = get_required_var("MITIGATION_GIT_REPO_PASS");
+my $git_branch_name = get_required_var("MITIGATION_GIT_BRANCH_NAME");
+my $deploy_script   = get_required_var("DEPLOY_SCRIPT");
+my $run_id          = get_required_var("RUN_ID");
+
+
+sub run {
+    my $self = shift;
+    #Prepare mitigations-testsuite.git
+    assert_script_run("git config --global http.sslVerify false");
+    assert_script_run("rm -rf mitigation-testsuite");
+    assert_script_run("git clone -q --single-branch -b $git_branch_name --depth 1 $git_repo_url");
+    assert_script_run("pushd mitigation-testsuite");
+    assert_script_run("git status");
+    assert_script_run("PAGER= git log -1");
+
+    #deploy script might be return not 0. use script_run.
+    #we ignore exit code now. testing result will be checked later.
+    script_run("sh deploy_scripts/${deploy_script}");
+
+    #Prepare config and list file.
+    assert_script_run("sed \'/SQ_ABUILD_PARTITION/d\' io-test-list >~/qaset/list");
+    assert_script_run("echo _QASET_RUNID=${run_id} >~/qaset/config");
+    assert_script_run("echo NPB_CLASS_SET=\"C\" >>~/qaset/config");
+
+    #check kernel if set mitigations=off.
+    #If yes, we set to default mode: mitigations=auto.
+    my $ret = script_run("grep \"mitigations=off\" /proc/cmdline");
```

Comment:
Sorry, I don't understand.

Reply:
I think you can wrap a function and put it into to lib/mitigation.pm

Reply:
I only saw the mitigations=auto situation perf test is handled , not saw the mitigations=on situation. Do we not need to collect data mitigations=on situation?

Reply:
I think we only handle "default_enable" now. 
mitigations=auto is "default_enable".
And If we want to handle more scenarios, I agree to extend it in future, not now.
Now we need a baseline of code so that other people could add features into.